### PR TITLE
feat: Attachment global errors

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**
Depends on #532

### CODE REVIEW

File: Mail/Views/AlertView.swift [hunk 1]
```diff
@@ -47,23 +61,69 @@ struct CustomAlertModifier<AlertContent>: ViewModifier where AlertContent: View
...
 
+private struct ClearFullScreenView: UIViewRepresentable {
+    private class BackgroundRemovalView: UIView {
+        override func didMoveToWindow() {
+            super.didMoveToWindow()
+            clearBackgroundSuperviews(view: self)
+        }
+
+        private func clearBackgroundSuperviews(view: UIView, level: Int = 0) {
+            guard level < 5 else {
```
Comment (adrien-coye) [inline Mail/Views/AlertView.swift:101]: Can we name the const to make more explicit what it does ?

File: Mail/Views/AlertView.swift [hunk 2]
```diff
@@ -47,23 +61,69 @@ struct CustomAlertModifier<AlertContent>: ViewModifier where AlertContent: View
...
+
+            if let superview = view.superview {
+                superview.backgroundColor = .clear
+                clearBackgroundSuperviews(view: superview, level: level + 1)
+            }
+        }
+    }
+
+    func makeUIView(context: Context) -> UIView {
+        return BackgroundRemovalView()
```
Comment (adrien-coye) [inline Mail/Views/AlertView.swift:113]: You are more into returning directly than I might be, but what is the prefered approach with implicit returns in closures ? I like to ommit retrun when possible.

File: Mail/Views/New Message/Attachments/AttachmentsManager.swift [hunk 3]
```diff
@@ -26,8 +26,12 @@ class AttachmentUploadTask: ObservableObject {
     @Published var progress: Double = 0
     var task: Task<String?, Never>?
     @Published var error: MailError?
+    var uploadDone: Bool {
+        return progress == 1
```
Comment (adrien-coye) [inline Mail/Views/New Message/Attachments/AttachmentsManager.swift:30]: It does not change anything, but I'd write >= 1.

**Review by adrien-coye — COMMENTED** [review]
I only had small questions. LGTM.

File: Mail/Views/AlertView.swift (hunk 1 above)
Comment (PhilippeWeidmann) [reply Mail/Views/AlertView.swift:101]: Resolved in dependant PR

**Review by Lnamw — DISMISSED** [review]

**Review by Ambrdctr — APPROVED** [review]

**Review by Lnamw — APPROVED** [review]

**Pull request merged**